arelle/ValidateFilingText.py

```diff
@@ -497,7 +501,8 @@ def close(self): pass
             if isInline and '/>' in line:
                 for match in inlineSelfClosedElementPattern.finditer(line):
                     selfClosedLocalName = match.group('localName')
-                    if selfClosedLocalName not in elementsWithNoContent:
+                    if (selfClosedLocalName not in elementsWithNoContent and
+                        not (selfClosedLocalName in ixElementsAllowingNil and xsiNilPattern.match(match.group('attributes') or ''))):
```

Comment:
I don't understand.  The point of this warning is that browsers in non-XHTML mode will interpret this:
```
<ix:nonNumeric/>
<div>content</div>
```
...as this:
```
<ix:nonNumeric><div>content</div></ix:nonNumeric>
```
It doesn't matter whether the element has `xsi:nil` or not, it's wrong to output self-closing.

For the specific case we discussed, the SEC has logic that rewrites the iXBRL, and that logic is wrongly introducing self-closing elements.  I think that logic needs to be fixed, not the warning.

Reply:
Completely agree with brettkail-wk.
The reserializer in arelle/EDGAR/render needs to be fixed, not this.